src/main/scala/codacy/scalastyle/ScalaStyle.scala

```diff
@@ -56,21 +56,22 @@ object ScalaStyle extends Tool {
 
   private lazy val scalaStyleConfig: Elem = XML.loadFile(defaultConfigFile)
 
-  private def parseToolResult(lines: List[String], source: Source.Directory): List[Result] = {
-
-    val RegMatch = """([a-z]+) file=(.+) id=(.+) message=(.+) line=([0-9]+).*$""".r
-    val FileErrorMatch = """^error file=(.+) message=(.+)""".r
-
-    lines.collect {
-      case RegMatch(level, file, id, message, line) =>
-        val patternId = id.split('.').last
-        val filePath = file
-        Result.Issue(Source.File(filePath), Result.Message(message), Pattern.Id(patternId), Source.Line(line.toInt))
-
-      case FileErrorMatch(file, message) =>
-        Result.FileError(Source.File(file), Option(ErrorMessage(message)))
+  private def parseToolResult(result: List[Message[FileSpec]], messageHelper: MessageHelper): List[Result] =
```

Comment:
Let's say it looks a bit better using the API instead of parsing the command line, good job kind sir!